src/view/uniprotkb/components/VariationView.tsx

```diff
@@ -46,17 +53,120 @@ interface ChangeEvent extends Event {
   detail?: { type: string; value: string[] };
 }
 
+const formatVariantDescription = (description: string) => {
+  /* eslint-disable no-useless-escape */
+  const pattern = /\[(\w+)\]: ([^\[]+)/g;
+  const match = description.match(pattern);
+  return match;
+};
+
 const VariationView: FC<{ primaryAccession: string }> = ({
   primaryAccession,
 }) => {
   loadWebComponent('protvista-variation', ProtvistaVariation);
   loadWebComponent('protvista-navigation', ProtvistaNavigation);
   loadWebComponent('protvista-sequence', ProtvistaSequence);
   loadWebComponent('protvista-manager', ProtvistaManager);
+  loadWebComponent('protvista-datatable', ProtvistaDatatable);
   loadWebComponent('protvista-filter', ProtvistaFilter);
   loadWebComponent('protvista-checkbox', ProtvistaCheckbox);
   loadWebComponent('protvista-variation-adapter', ProtvistaVariationAdapter);
 
+  const [showEvidenceTagData, setShowEvidenceTagData] = useState(false);
+  const [selectedEvidenceData, setSelectedEvidenceData] = useState();
+  const [selectedReferences, setSelectedReferences] = useState();
+
+  const evidenceTagCallback = (
+    evidenceData: EvidenceData,
+    references: Evidence[] | undefined
+  ) => {
+    setSelectedEvidenceData(evidenceData);
+    setSelectedReferences(references);
+    setShowEvidenceTagData(true);
+  };
+
+  const columns = {
```

Comment:
The `columns` variable doesn't change between successive evaluations of `VariationView` but takes up several of the component's lines. A suggestion is to take this out of the component:
```
const getColumns = (
  evidenceTagCallback: (
    evidenceData: EvidenceData,
    references: Evidence[] | undefined
  ) => void
) => ({
  positions: {
    label: 'Position(s)',
    resolver: (d: ProtvistaVariant) =>
      d.begin === d.end ? d.begin : `${d.begin}-${d.end}`,
  },
...
```
and then `setTableData` will only depend on `data`:
```
const setTableData = useCallback(
    (node): void => {
      if (node) {
        // eslint-disable-next-line no-param-reassign
        node.data = data.features;

        const evidenceTagCallback = (
          evidenceData: EvidenceData,
          references: Evidence[] | undefined
        ) => {
          setSelectedEvidenceData(evidenceData);
          setSelectedReferences(references);
          setShowEvidenceTagData(true);
        };

        // eslint-disable-next-line no-param-reassign
        node.columns = getColumns(evidenceTagCallback);
      }
    },
    [data]
  );
```